From: Tal Zussman <tz2294@columbia.edu>
Subject: compiler_types: remove unnecessary indirection in compiletime_assert()
Date: Sat, 07 Jun 2025 03:23:55 -0400

compiletime_assert() is the only user of _compiletime_assert(), which uses
__compiletime_assert() directly with no modification to arguments.  Remove
_compiletime_assert() and use __compiletime_assert() directly.

Link: https://lkml.kernel.org/r/20250607-compiletime_assert-v1-1-a8991a9be729@columbia.edu
Signed-off-by: Tal Zussman <tz2294@columbia.edu>
Cc: Kees Cook <kees@kernel.org>
Cc: Miguel Ojeda <ojeda@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/compiler_types.h |    5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

--- a/include/linux/compiler_types.h~compiler_types-remove-unnecessary-indirection-in-compiletime_assert
+++ a/include/linux/compiler_types.h
@@ -552,9 +552,6 @@ struct ftrace_likely_data {
 # define __compiletime_assert(condition, msg, prefix, suffix) ((void)(condition))
 #endif
 
-#define _compiletime_assert(condition, msg, prefix, suffix) \
-	__compiletime_assert(condition, msg, prefix, suffix)
-
 /**
  * compiletime_assert - break build and emit msg if condition is false
  * @condition: a compile-time constant condition to check
@@ -565,7 +562,7 @@ struct ftrace_likely_data {
  * compiler has support to do so.
  */
 #define compiletime_assert(condition, msg) \
-	_compiletime_assert(condition, msg, __compiletime_assert_, __COUNTER__)
+	__compiletime_assert(condition, msg, __compiletime_assert_, __COUNTER__)
 
 #define compiletime_assert_atomic_type(t)				\
 	compiletime_assert(__native_word(t),				\
_
